### PR TITLE
Preserve legacy inflight metrics and fixes registration 

### PR DESCRIPTION
**Pull request by yue9944882**
/sig api-machinery
/kind feature

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by RainbowMango** [comment]
Can you add a `Fixes #88426` to the PR descriptions?
I see you have fixed all items in #88426.

File: staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness.go [hunk 1]
```diff
@@ -57,6 +55,8 @@ func GetClassification(ctx context.Context) *PriorityAndFairnessClassification {
 	return ctx.Value(priorityAndFairnessKey).(*PriorityAndFairnessClassification)
 }
 
+var atomicMutatingLen, atomicNonMutatingLen = atomic.NewInt32(0), atomic.NewInt32(0)
```
Comment (MikeSpreitzer) [inline staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness.go]: Why use a third party package (go.uber.org/atomic) when we can get the functionality we are using here from sync/atomic?

**Review by MikeSpreitzer — COMMENTED** [review]
I would say that the change here is not "fixes registration" but rather "uses the standard metrics abstraction instead of going directly to Prometheus".

**Comment by MikeSpreitzer** [comment]
@MikeSpreitzer

**Review by logicalhan — APPROVED** [review]
/lgtm

**Comment by logicalhan** [comment]
Need to run golint though.

**Comment by MikeSpreitzer** [comment]
And we agreed to replace the use of uber's atomic with golang's atomic.

**Comment by MikeSpreitzer** [comment]
@yue9944882 : you have just a couple of tiny things to do here.

**Review by MikeSpreitzer — APPROVED** [review]
/lgtm

**Pull request merged**